pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -187,6 +261,280 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
 	return r.ensureOutputObject(instance, tpcm, reqLogger)
 }
 
+// handleMigration checks if the TailoredProfile has any deprecated variables or any updated KubeletConfig rules
+// and handle the migration
+func (r *ReconcileTailoredProfile) handleMigration(
+	v1alphaTp *cmpv1alpha1.TailoredProfile, logger logr.Logger) (bool, error) {
+
+	// Get the list of KubeletConfig rules that are migrated with checkType change
+	migratedRules, err := r.getMigratedRules(v1alphaTp, logger)
+	if err != nil {
+		return false, err
+	}
+
+	if len(migratedRules) == 0 {
+		return true, nil
+	}
+
+	profileType := utils.GetScanType(v1alphaTp.GetAnnotations())
+
+	doContinue := true
+	v1alphaTpCP := v1alphaTp.DeepCopy()
+
+	// check if there are any disabled rules that are migrated
+	if len(v1alphaTp.Spec.DisableRules) > 0 {
+		var newRules []cmpv1alpha1.RuleReferenceSpec
+		for ri := range v1alphaTp.Spec.DisableRules {
+			rule := &v1alphaTp.Spec.DisableRules[ri]
+			if checkType, ok := migratedRules[rule.Name]; ok && checkType != string(profileType) {
+				// remove the rule from the list of disabled rules
```

Comment:
Talking myself through this - but technically leaving these rules in the Tailored Profile wouldn't break anything. However, it gives the false impression that they are doing something, when they're not doing anything at all since they're now the wrong type.

Reply:
that's correct, so we just remove those from the tailored profile